### PR TITLE
Docs and examples

### PR DESCRIPTION
**Pull request by NaokiHamada**
- [x] Add `matplotlib.use('Agg')` and remove `plt.show()` in communication example scripts for headless/CI compatibility
- [x] Fix 3-objective facility location doc to restate all demand-center coordinates
- [x] Add `test_splitting_module_public_api` to `tests/splitting.py`
- [x] Add `pathlib.Path("docs/_static").mkdir(...)` before savefig in all example scripts
- [x] Fix simplex terminology: "3-simplex" → "2-simplex" in 3-objective example scripts and docs
- [x] Fix simplex terminology: "4-simplex" → "3-simplex" in 4-objective section of `facility_location.rst`
- [x] Add numerical experiments to `portfolio.rst`, `federated_learning.rst`, `medical_imaging.rst`, `model_predictive_control.rst`, `smart_grids.rst`, `supply_chain.rst`
- [x] Fix "Bezier" → "Bézier" in `generate_communication_pareto_3obj.py` legend label and in comments in `generate_communication_pareto.py`
- [x] Add Pareto set figures to all application example scripts
- [x] Update all application RST docs to side-by-side layout
- [x] Fix `medical_imaging.rst` Problem Setup to match the actual example script
- [x] Fix "Bezier" → "Bézier" in comments in `generate_supply_chain_pareto.py`, `generate_smart_grids_pareto.py`, `generate_portfolio_pareto.py`, `generate_mpc_pareto.py`, `generate_federated_learning_pareto.py`, `generate_communication_pareto.py`, `generate_medical_imaging_pareto.py`

### CODE REVIEW

File: docs/advanced/splitting.rst [hunk 1]
```diff
@@ -341,9 +341,9 @@ API Reference
...
 for the full parameter reference:
 
-* :func:`torch_bsf.split`
-* :func:`torch_bsf.reparametrize`
-* :func:`torch_bsf.longest_edge_criterion`
-* :func:`torch_bsf.max_error_criterion`
-* :func:`torch_bsf.split_by_criterion`
-* :data:`torch_bsf.SplitCriterion`
+* :func:`torch_bsf.splitting.split`
+* :func:`torch_bsf.splitting.reparametrize`
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply docs/advanced/splitting.rst]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3035586888)

File: examples/generate_communication_pareto.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+# Calculate approximation error
+errors = np.sqrt((y_smooth[:, 0] - np.interp(t_smooth, weights, f1_targets))**2 +
+                 (y_smooth[:, 1] - np.interp(t_smooth, weights, f2_targets))**2)
+max_error = np.max(errors)
+print(f'Max approximation error: {max_error:.4f}')
+
+plt.tight_layout()
+plt.savefig('docs/_static/communication_fitting.png', dpi=150, bbox_inches='tight')
+plt.show()
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_communication_pareto.py]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036019376)

File: examples/generate_communication_pareto_3obj.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ax.scatter(predicted[:, 0], predicted[:, 1], predicted[:, 2], color="red", alpha=0.15, s=10, label="Bezier simplex approximation")
+ax.set_xlabel("f₁: Congestion Excess")
+ax.set_ylabel("f₂: Route Volatility")
+ax.set_zlabel("f₃: Path Imbalance")
+ax.set_title("Three-objective Pareto Front: Optimization vs Bézier Simplex")
+ax.legend()
+
+plt.tight_layout()
+plt.savefig("docs/_static/communication_pareto_3obj.png", dpi=150, bbox_inches="tight")
+plt.show()
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_communication_pareto_3obj.py]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036019381)

File: examples/generate_facility_location_pareto_2obj.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+# Prepare training data for Bézier simplex fitting
+w1 = np.array([p[0][0] for p in optimals])
+w2 = np.array([p[0][1] for p in optimals])
+w3 = np.zeros_like(w1)
+X = np.column_stack([w1, w2, w3])
+y = np.column_stack([np.array([p[2] for p in optimals]), np.array([p[3] for p in optimals])])
+
+# Fit a degree-3 Bézier simplex to the Pareto front
+regressor = BezierSimplexRegressor(degree=3)
+regressor.fit(X, y)
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_facility_location_pareto_2obj.py:46]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036019374)

File: docs/applications/facility_location.rst [hunk 5]
```diff
@@ -14,3 +14,58 @@ Likewise, in engineering design optimizing for structural compliance or minimum
...
+         :width: 100%
+     - .. image:: ../_static/facility_location_pareto_2obj.png
+         :alt: Bézier simplex fitting to two-objective facility location Pareto front
+         :width: 100%
+   * - Pareto set (2-obj).
+     - Pareto front (2-obj).
+
+**Three Objectives:**
+We add a third demand center at :math:`a_3=(0.5,1)`, with :math:`f_3(x) = \lVert x - a_3\rVert^2`. The weighted objective becomes :math:`f(x, w) = w_1 f_1(x) + w_2 f_2(x) + w_3 f_3(x)`. We sample 10 weight vectors on the 3-simplex and fit a degree-3 Bézier simplex.
+
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply docs/applications/facility_location.rst:41]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036019366)

File: tests/splitting.py [hunk 6]
```diff
@@ -485,20 +485,3 @@ def test_split_by_criterion_max_error():
     bs_A, bs_B = split_by_criterion(bs, criterion)
     assert bs_A.degree == bs.degree
     assert bs_B.degree == bs.degree
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply tests/splitting.py:487]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036049731)

File: examples/generate_communication_pareto_3obj.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        [0.0, 0.0, 1.0],
+        [0.5, 0.5, 0.0],
+        [0.5, 0.0, 0.5],
+        [0.0, 0.5, 0.5],
+        [1 / 3, 1 / 3, 1 / 3],
+        [0.7, 0.2, 0.1],
+        [0.2, 0.7, 0.1],
+        [0.1, 0.2, 0.7],
+    ]
+)
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_communication_pareto_3obj.py]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036424963)

File: examples/generate_facility_location_pareto_3obj.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    [0.0, 1.0, 0.0],
+    [0.0, 0.0, 1.0],
+    [0.5, 0.5, 0.0],
+    [0.5, 0.0, 0.5],
+    [0.0, 0.5, 0.5],
+    [1/3, 1/3, 1/3],
+    [0.7, 0.2, 0.1],
+    [0.2, 0.7, 0.1],
+    [0.1, 0.2, 0.7],
+])
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_facility_location_pareto_3obj.py]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036424958)

[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_facility_location_pareto_3obj.py]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036424958)

**Comment by NaokiHamada** [comment]
@copilot facility_location.rst と communication.rst 以外の application にも数値例を追加して。

File: docs/applications/facility_location.rst [hunk 9]
```diff
@@ -14,3 +14,58 @@ Likewise, in engineering design optimizing for structural compliance or minimum
...
+         :width: 100%
+     - .. image:: ../_static/facility_location_pareto_3obj.png
+         :alt: Bézier simplex fitting to three-objective facility location Pareto front
+         :width: 100%
+   * - Pareto set (3-obj).
+     - Pareto front (3-obj).
+
+**Four Objectives:**
+We add a fourth demand center at :math:`a_4=(1,1)`, with :math:`f_4(x) = \lVert x - a_4\rVert^2`. The weighted objective is :math:`f(x, w) = w_1 f_1(x) + w_2 f_2(x) + w_3 f_3(x) + w_4 f_4(x)`. We sample 10 weight vectors on the 4-simplex and fit a degree-2 Bézier simplex. For visualization, we project to 3D by plotting :math:`f_1, f_2, f_3`.
+
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply docs/applications/facility_location.rst:57]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036576468)

File: examples/generate_communication_pareto_3obj.py [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+# Plot 3D Pareto front approximation
+fig = plt.figure(figsize=(9, 7))
+ax = fig.add_subplot(111, projection="3d")
+ax.scatter(y[:, 0], y[:, 1], y[:, 2], color="blue", label="Optimization-derived points", s=40)
+ax.scatter(predicted[:, 0], predicted[:, 1], predicted[:, 2], color="red", alpha=0.15, s=10, label="Bezier simplex approximation")
+ax.set_xlabel("f₁: Congestion Excess")
+ax.set_ylabel("f₂: Route Volatility")
+ax.set_zlabel("f₃: Path Imbalance")
+ax.set_title("Three-objective Pareto Front: Optimization vs Bézier Simplex")
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_communication_pareto_3obj.py:88]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036576492)

**Review by NaokiHamada — COMMENTED** [review]
@copilot すべての application について、pareto set にフィッティングしたベジエ単体の図も追加して、既にある pareto front にフィッティングしたベジエ単体の図と side by side で並べて。

File: examples/generate_communication_pareto.py [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    optimal_points.append((w, x_opt, f1_val, f2_val))
+
+# Extract data for fitting
+w1_vals = np.array([p[0][0] for p in optimal_points])
+w2_vals = np.array([p[0][1] for p in optimal_points])
+f1_targets = np.array([p[2] for p in optimal_points])
+f2_targets = np.array([p[3] for p in optimal_points])
+
+# Fit Bezier simplex (multi-output: predict f1 and f2)
+X = np.column_stack([w1_vals, w2_vals])
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_communication_pareto.py]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3036576501)

**Comment by NaokiHamada** [comment]
@copilot 多くの数値例で、軸のスケールが異なるために、ベジエ単体フィッティングが暴れている。'normalize=max' オプションを使ってスケールを揃えて。

**Comment by NaokiHamada** [comment]
@copilot ごめん、さっきの指示が悪かった。指示の前のコミットにrevertして。

File: docs/applications/medical_imaging.rst [hunk 12]
```diff
@@ -21,3 +21,44 @@ where the first component penalizes dose errors via beamlet fluence :math:`f`, a
...
+   f_1(x) &= \|Kx - y\|^2 + 0.01\|x\|^2 \quad \text{(fidelity)} \\
+   f_2(x) &= \|Lx\|^2 + 0.01\|x\|^2 \quad \text{(smoothness)}
+
+**Experiment Procedure:**
+
+1. Sample 10 weight vectors :math:`w = (w_1, w_2)` on the 1-simplex from :math:`(1,0)` to :math:`(0,1)`.
+2. For each :math:`w`, solve :math:`x^*(w) = \arg\min_x [w_1 f_1(x) + w_2 f_2(x)]` using L-BFGS-B.
+3. Collect the Pareto front points :math:`(f_1(x^*(w)), f_2(x^*(w)))`.
+4. Fit a degree-3 Bézier simplex to the weight–objective pairs.
+5. Visualize the fitted Bézier curve against the L-curve sampled by optimization.
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply docs/applications/medical_imaging.rst]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3042790766)

File: examples/generate_supply_chain_pareto.py [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+w1 = np.array([p[0][0] for p in optimals])
+w2 = np.array([p[0][1] for p in optimals])
+X = np.column_stack([w1, w2])
+y = np.column_stack([
+    np.array([p[2] for p in optimals]),
+    np.array([p[3] for p in optimals]),
+])
+
+# Fit a degree-3 Bezier simplex to the Pareto front
+regressor = BezierSimplexRegressor(degree=3)
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_supply_chain_pareto.py]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3042790796)

File: examples/generate_communication_pareto.py [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+y = np.column_stack([f1_targets, f2_targets])
+regressor = BezierSimplexRegressor(degree=3)
+regressor.fit(X, y)
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply examples/generate_communication_pareto.py:52]: @copilot apply changes based on [this feedback](https://github.com/opthub-org/pytorch-bsf/pull/328#discussion_r3042790850)

File: torch_bsf/sklearn.py [hunk 15]
```diff
@@ -52,12 +52,12 @@ class BezierSimplexRegressor(BaseEstimator, RegressorMixin):
...
         self,
-        degree: int = 3,
+        degree: int | None = None,
         smoothness_weight: float = 0.0,
         init: BezierSimplex | ControlPointsData | None = None,
         freeze: Iterable[Index] | None = None,
         batch_size: int | None = None,
-        max_epochs: int = 1000,
+        max_epochs: int = 3,
         accelerator: str = "auto",
```
[the remark this answers is not in the document]
Comment (NaokiHamada) [reply torch_bsf/sklearn.py:62]: @copilot do not restore. That was a correct change.
